engine/flink/kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/source/flink/FlinkKafkaSource.scala

```diff
@@ -127,6 +128,11 @@ class FlinkKafkaSource[T](
     deserializeTestData(formatter.parseRecord(topics.head, testParametersInfo.createTestRecord(flatParams)))
   }
 
+  private def prepareConsumerGroupId(nodeContext: FlinkCustomNodeContext): String = {
+    val baseName = overriddenConsumerGroup.getOrElse(ConsumerGroupDeterminer(kafkaConfig).consumerGroup(nodeContext))
+    namingStrategy.prepareName(baseName)
```

Comment:
We have to add a feature flag that allow smooth migration. I would pick the "opt-out" approach in this case. I mean: we add `KafkaConfig.useNamingStrategyInConsumerGroups = true`, we ensure that before the deployment of the new version, we set it to `false` in the config file. After we do the migration, we can remove it from the config and from the code as well.

Reply:
done :+1: I've added an entry in kafka configuration docs and in migration guide about the feature flag. I've named it `useNamingStrategyForConsumerGroupId`